cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -386,50 +413,57 @@ synchronized UserTaskInfo getUserTaskByUserTaskId(UUID userTaskId, HttpServletRe
   private synchronized UserTaskInfo insertFuturesByUserTaskId(UUID userTaskId,
                                                               Function<String, OperationFuture> operation,
                                                               HttpServletRequest httpServletRequest) {
-    if (_allUuidToUserTaskInfoMap.get(TaskState.COMPLETED).containsKey(userTaskId)) {
-      // Before add new operation to task, first recycle the task from completed task list.
-      _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).put(userTaskId, _allUuidToUserTaskInfoMap.get(TaskState.COMPLETED)
-          .remove(userTaskId).setState(TaskState.ACTIVE));
-      LOG.info("UserTask {} is recycled from complete task list and added back to active tasks list", userTaskId);
-    }
-    if (_allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).containsKey(userTaskId)) {
-      _allUuidToUserTaskInfoMap.get(TaskState.ACTIVE).get(userTaskId).futures().add(operation.apply(userTaskId.toString()));
+    for (Map<UUID, UserTaskInfo> infoMap : _uuidToCompletedUserTaskInfoMap.values()) {
+      if (infoMap.containsKey(userTaskId)) {
+        // Before add new operation to task, first recycle the task from completed task list.
+        _uuidToActiveUserTaskInfoMap.put(userTaskId, infoMap.remove(userTaskId).setState(TaskState.ACTIVE));
+        LOG.info("UserTask {} is recycled from complete task list and added back to active tasks list", userTaskId);
```

Comment:
Do we still need such a recycling functionality supported or can we get rid of it?